server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateUpdater.java

```diff
@@ -77,9 +72,9 @@ public QualityGateDto copy(DbSession dbSession, QualityGateDto qualityGateDto, S
     return destinationGate;
   }
 
-  private void validateQualityGateUpdate(DbSession dbSession, Long qualityGateId, String name) {
+  private void validateQualityGate(DbSession dbSession, OrganizationDto organizationDto, String name) {
     List<String> errors = new ArrayList<>();
-    checkQualityGateDoesNotAlreadyExist(dbSession, qualityGateId, name, errors);
+    checkQualityGateDoesNotAlreadyExist(dbSession, organizationDto, name, errors);
     checkRequest(errors.isEmpty(), errors);
```

Comment:
Please remove checkRequest => it's no more needed as there's now only one error that can be generated in `checkQualityGateDoesNotAlreadyExist`.
Moreoever, `checkQualityGateDoesNotAlreadyExist` should then directly throw an IllegalArgumentException instead of adding an error to the list of error.

Reply:
done